### PR TITLE
Tweaks to influx

### PR DESCRIPTION
**Pull request by i3149**
Messing with influx to change how attributes are set. 

```
kentik.SYNOLOGY-SYSTEM-MIB,device_ip=10.10.0.251,modelName=DS1520+,device_name=mabel,objectIdentifier=1.3.6.1.4.1.6574.1.3.0,instrumentation.name=disk_station,upgradeAvailable=available,systemStatus=normal,systemFanStatus=normal,powerStatus=normal,version=DSM\ 7.0-42218,serialNumber=2110RYRMQ9AXB,tags.type=foo,cpuFanStatus=normal powerStatus=1i 1658869145000000000
kentik.SYNOLOGY-SYSTEM-MIB,objectIdentifier=1.3.6.1.4.1.6574.1.1.0,device_name=mabel,upgradeAvailable=available,tags.type=foo,version=DSM\ 7.0-42218,systemStatus=normal,modelName=DS1520+,device_ip=10.10.0.251,instrumentation.name=disk_station,serialNumber=2110RYRMQ9AXB,systemFanStatus=normal,powerStatus=normal,cpuFanStatus=normal systemStatus=1i 1658869145000000000
kentik.SYNOLOGY-SYSTEM-MIB,modelName=DS1520+,serialNumber=2110RYRMQ9AXB,instrumentation.name=disk_station,upgradeAvailable=available,systemStatus=normal,objectIdentifier=1.3.6.1.4.1.6574.1.5.4.0,systemFanStatus=normal,cpuFanStatus=normal,tags.type=foo,version=DSM\ 7.0-42218,device_ip=10.10.0.251,powerStatus=normal,device_name=mabel upgradeAvailable=1i 1658869145000000000
kentik.SYNOLOGY-SYSTEM-MIB,modelName=DS1520+,version=DSM\ 7.0-42218,systemFanStatus=normal,device_name=mabel,cpuFanStatus=normal,serialNumber=2110RYRMQ9AXB,upgradeAvailable=available,tags.type=foo,device_ip=10.10.0.251,objectIdentifier=1.3.6.1.4.1.6574.1.2.0,systemStatus=normal,powerStatus=normal,instrumentation.name=disk_station temperature=37i 1658869145000000000
kentik.SYNOLOGY-DISK-MIB,device_name=mabel,objectIdentifier=1.3.6.1.4.1.6574.2.1.1.5,instrumentation.name=disk_station,device_ip=10.10.0.251,diskModel=ST12000VN0008-2PH103,tags.type=foo,diskID=Disk\ 4,diskType=SATA,index=.0,mib-table=disk,diskStatus=normal diskStatus=1i 1658869145000000000
kentik.SYNOLOGY-DISK-MIB,index=.0,device_name=mabel,device_ip=10.10.0.251,diskModel=ST12000VN0008-2PH103,diskType=SATA,objectIdentifier=1.3.6.1.4.1.6574.2.1.1.6,instrumentation.name=disk_station,mib-table=disk,diskStatus=normal,tags.type=foo,diskID=Disk\ 4 diskTemperature=37i 1658869145000000000
kentik.SYNOLOGY-DISK-MIB,device_ip=10.10.0.251,mib-table=disk,objectIdentifier=1.3.6.1.4.1.6574.2.1.1.5,instrumentation.name=disk_station,tags.type=foo,diskType=SATA,device_name=mabel,diskModel=ST12000VN0008-2PH103,diskID=Disk\ 5,diskStatus=notinitialized,index=.1 diskStatus=3i 1658869145000000000
kentik.SYNOLOGY-DISK-MIB,mib-table=disk,diskID=Disk\ 5,diskStatus=notinitialized,diskType=SATA,index=.1,instrumentation.name=disk_station,device_ip=10.10.0.251,diskModel=ST12000VN0008-2PH103,tags.type=foo,objectIdentifier=1.3.6.1.4.1.6574.2.1.1.6,device_name=mabel diskTemperature=35i 1658869145000000000
kentik.SYNOLOGY-DISK-MIB,device_ip=10.10.0.251,instrumentation.name=disk_station,tags.type=foo,diskID=Disk\ 1,diskType=SATA,device_name=mabel,diskModel=WD120EFBX-68B0EN0,objectIdentifier=1.3.6.1.4.1.6574.2.1.1.6,index=.2,mib-table=disk,diskStatus=normal diskTemperature=36i 1658869145000000000
```

### CODE REVIEW

File: pkg/formats/influx/influx.go [hunk 1]
```diff
@@ -30,6 +31,10 @@ type InfluxData struct {
 	Timestamp   int64
 }
 
+var (
+	Prefix = flag.String("influxdb_measurment_prefix", "kentik.", "Prefix metric names with this")
```
Comment (tdanner) [inline pkg/formats/influx/influx.go:35]: s/measurment/measurement/

Comment (tdanner) [reply pkg/formats/influx/influx.go:35]: Also, do we need a default "kentik." prefix?

File: pkg/formats/influx/influx.go [hunk 2]
```diff
@@ -357,3 +364,39 @@ func influxEscape(s string) string {
...
+	return mib
+}
+
+var dropFields = map[string]bool{
+	//"device_name": true,
+	"eventType":   true,
+	"SysObjectID": true,
+	"provider":    true,
+	"SysServices": true,
+	"SysDescr":    true,
```
Comment (tdanner) [inline pkg/formats/influx/influx.go:401]: We don't need SysDescr and SysObjectID to be reported in every line, but we do need them reported sometime or occasionally. Not sure what the best way to manage that would be.

Comment (i3149) [reply pkg/formats/influx/influx.go:401]: In NR these are attached to uptime metrics and only these.

**Review by tdanner — DISMISSED** [review]
I'm sure we'll have more adjustments, but this looks like a big improvement to me.

**Comment by dpajin-kentik** [comment]
@i3149, @tdanner  I think we should: 
- remove "objectIdentifier" from the tags and combine all the "metrics" with the same tags into the same line. 
- Also, the measurement is using MIB name, but it should be "Table name". Thinking a bit more on this, maybe at the point it should be "MIB name" + "::" + "Table name" to ensure uniqueness across the board. In this example, the measurement should be "SYNOLOGY-DISK-MIB::diskTable" (based on the profile here: https://github.com/kentik/snmp-profiles/blob/main/profiles/kentik_snmp/synology/disk_station.yml)
- The tag named "mib-table" is redundant, so we shoudl remove it.

For example, these two lines:
```
kentik.SYNOLOGY-DISK-MIB,device_ip=10.10.0.251,mib-table=disk,objectIdentifier=1.3.6.1.4.1.6574.2.1.1.5,instrumentation.name=disk_station,tags.type=foo,diskType=SATA,device_name=mabel,diskModel=ST12000VN0008-2PH103,diskID=Disk\ 5,diskStatus=notinitialized,index=.1 diskStatus=3i 1658869145000000000

kentik.SYNOLOGY-DISK-MIB,mib-table=disk,diskID=Disk\ 5,diskStatus=notinitialized,diskType=SATA,index=.1,instrumentation.name=disk_station,device_ip=10.10.0.251,diskModel=ST12000VN0008-2PH103,tags.type=foo,objectIdentifier=1.3.6.1.4.1.6574.2.1.1.6,device_name=mabel diskTemperature=35i 1658869145000000000
```
has the same set of tags (except objectIdentifier), with the same index from the table, so they should be combined into one line with two metrics, like this:

```
kentik.SYNOLOGY-DISK-MIB::diskTable,device_ip=10.10.0.251,instrumentation.name=disk_station,tags.type=foo,diskType=SATA,device_name=mabel,diskModel=ST12000VN0008-2PH103,diskID=Disk\ 5,diskStatus=notinitialized,index=.1 diskStatus=3i,diskTemperature=35i 1658869145000000000
```

where does "instrumentation.name=disk_station" tag comes from, same for "tags.type=foo"?

**Comment by tdanner** [comment]
Dusan is absolutely right - these changes would make the output way more usable.

**Comment by dpajin-kentik** [comment]
a few more things: 

- for metrics which are enum type, we would need to put the metric's "string" value, also in the Influx fields and not tags, with the metric's name with the additional suffix "_str". For example:
```
kentik.SYNOLOGY-SYSTEM-MIB,device_ip=10.10.0.251,modelName=DS1520+,device_name=mabel,version=DSM\ 7.0-42218,serialNumber=2110RYRMQ9AXB,tags.type=foo powerStatus=1i,cpuFanStatus=1i,upgradeAvailable=1i,systemStatus=1i,systemFanStatus=1i,powerStatus_string=normal,cpuFanStatus_string=normal,upgradeAvailable_string=available,systemStatus_string=normal,systemFanStatus_string=normal 1658869145000000000
```

- the previous point we should do also with the "metrics_tags" from the SNMP profiles so that they are placed in the InfluxDB line format at tags and not fields

- for metrics which are scalar, as those from SYNOLOGY-SYSTEM-MIB, we should use "MIB Name" as the measurement name, which is done, but we should also combine all the scalar metrics from the same MIB into one line submitted to the InfluxDB. The example for the previous point is describing this requirement

**Comment by i3149** [comment]
```
SYNOLOGY-DISK-MIB::disk,device_ip=10.10.0.251,device_name=mabel,diskID=Disk\ 4,diskModel=ST12000VN0008-2PH103,diskStatus=normal,diskType=SATA,index=.0,instrumentation.name=disk_station,tags.type=foo diskStatus=1i,diskTemperature=36i 1658947212000000000
none,SysLocation=Unknown,SysName=mabel,device_ip=10.10.0.251,device_name=mabel,instrumentation.name=disk_station,modelName=DS1520+,serialNumber=2110RYRMQ9AXB,src_addr=10.10.0.251,tags.type=foo,version=DSM\ 7.0-42218 Uptime=591854i 1658947212000000000
IF-MIB::if,device_name=mabel,if_Address=169.254.11.80,if_AdminStatus=up,if_Description=ovs_eth0,if_Index=8,if_Netmask=255.255.0.0,if_PhysAddress=00:11:32:ee:8f:29,if_Speed=0,if_Type=ethernetCsmacd,if_interface_name=ovs_eth0,instrumentation.name=disk_station,tags.type=foo ifOutDiscards=0i,ifHCOutBroadcastPkts=0i,ifHCInMulticastPkts=0i,ifHCInBroadcastPkts=0i,ifHCOutMulticastPkts=0i,ifInDiscards=0i,if_AdminStatus=1i,ifHCInUcastPkts=0i,ifHCOutUcastPkts=0i 1658947212000000000
IF-MIB::if,device_name=mabel,if_AdminStatus=up,if_Description=eth0,if_Index=5,if_PhysAddress=00:11:32:ee:8f:29,if_Speed=0,if_Type=ethernetCsmacd,if_interface_name=eth0,instrumentation.name=disk_station,tags.type=foo ifHCInUcastPkts=0i,ifInDiscards=0i,ifHCInBroadcastPkts=0i,ifHCInMulticastPkts=0i,ifHCOutMulticastPkts=0i,ifHCOutUcastPkts=0i,ifOutDiscards=0i,ifHCOutBroadcastPkts=0i,if_AdminStatus=1i 1658947212000000000
SYNOLOGY-SYSTEM-MIB,cpuFanStatus=normal,device_ip=10.10.0.251,device_name=mabel,instrumentation.name=disk_station,modelName=DS1520+,powerStatus=normal,serialNumber=2110RYRMQ9AXB,systemFanStatus=normal,systemStatus=normal,tags.type=foo,upgradeAvailable=available,version=DSM\ 7.0-42218 systemStatus=1i,cpuFanStatus=1i,powerStatus=1i,temperature=36i,upgradeAvailable=1i,systemFanStatus=1i 1658947212000000000
```

**Comment by tdanner** [comment]
Nice update! I think we can come up with a better measurement name than `none` for the device-level scalars though. @dpajin-kentik recommended the MIB name, which makes sense to me.

**Comment by i3149** [comment]
>  "instrumentation.name=disk_station" tag comes from, same for "tags.type=foo"?

`instrumentation.name` is an alias for provider. 

`tags.*` are the tagging system. Users can define tags per device or globally and then these come in as attributes on all metrics. See https://github.com/kentik/ktranslate/pull/358 for where this feature is heading.

**Comment by dpajin-kentik** [comment]
> none,SysLocation=Unknown,SysName=mabel,device_ip=10.10.0.251,device_name=mabel,instrumentation.name=disk_station,modelName=DS1520+,serialNumber=2110RYRMQ9AXB,src_addr=10.10.0.251,tags.type=foo,version=DSM\ 7.0-42218 Uptime=591854i 1658947212000000000

I think that "none" measurement should be probably called "device" ?

**Comment by dpajin-kentik** [comment]
looks good now

**Review by tdanner — APPROVED** [review]

**Pull request merged**